### PR TITLE
Add EditSerieViewModel and Refactor with Shared Base Class

### PR DESCRIPTION
**Pull request by MathieuPFR**
## Objective
Implement the missing edit functionality for Series and eliminate code duplication between `CreateSerieViewModel` and `EditSerieViewModel` by introducing a shared base class, following the same architectural pattern established for Event ViewModels.

### New Files
- **`EditSerieViewModel.kt`**: Complete implementation for editing existing series
- **`BaseSerieFormViewModel.kt`**: Abstract base class containing all shared validation logic for Serie forms

### Refactored Files
- **`CreateSerieViewModel`**: Now extends `BaseSerieFormViewModel`
- **`EditSerieViewModel`**: Built to extend `BaseSerieFormViewModel` from the start

### Tests Added
- **`EditSerieViewModelTest.kt`** 
  - Load serie tests
  - Field validation tests
  - Update serie tests
  - Error handling tests
  
- **`BaseSerieFormViewModelTest.kt`** 
  - Shared validation logic tests
  - Date and time validation (including edge cases)
  - Error clearing tests
  - Integration tests

## Benefits
- **Reduced Code Duplication**: Validation logic is centralized in one place
- **Improved Maintainability**: Changes to validation rules only need to be made once
- **Consistent Architecture**: Serie ViewModels now follow the same pattern as Event ViewModels
- **Enhanced Test Coverage**: Comprehensive tests ensure reliability of both shared and specific functionality

## Testing
All tests pass successfully:
- CreateSerieViewModelTest (existing)
- EditSerieViewModelTest (new)
- BaseSerieFormViewModelTest (new)

Lot of lines added but only ~500 lines of real code, rest is tests
Next I will implement the UI of EditSerie

### CODE REVIEW

File: app/src/main/java/com/android/joinme/ui/overview/CreateSerieViewModel.kt [hunk 1]
```diff
@@ -166,178 +147,16 @@ class CreateSerieViewModel(
     return try {
       repository.addSerie(serie)
       clearErrorMsg()
-      _uiState.value = _uiState.value.copy(isLoading = false)
+      setLoadingState(false)
       true
     } catch (e: Exception) {
       Log.e("CreateSerieViewModel", "Error creating serie", e)
```
Comment (Atilla13) [inline app/src/main/java/com/android/joinme/ui/overview/CreateSerieViewModel.kt]: I would suggest to remove this Log statement and instead use setErrorMsg.

File: app/src/main/java/com/android/joinme/ui/overview/BaseSerieFormViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,351 @@
...
+            currentState.invalidMaxParticipantsMsg == null &&
+            currentState.invalidDateMsg == null &&
+            currentState.invalidTimeMsg == null &&
+            currentState.invalidVisibilityMsg == null &&
+            currentState.title.isNotBlank() &&
+            currentState.description.isNotBlank() &&
+            currentState.maxParticipants.isNotBlank() &&
+            currentState.date.isNotBlank() &&
+            currentState.time.isNotBlank() &&
+            currentState.visibility.isNotBlank()
```
[the remark this answers is not in the document]
Comment (Atilla13) [reply app/src/main/java/com/android/joinme/ui/overview/BaseSerieFormViewModel.kt]: Fair point about consistency. Let's keep it as is for now. I'll create an issue to track refactoring the validation logic across all components so we can improve this systematically later. Approving!

**Review by Atilla13 — APPROVED** [review]
Approved! I've created an issue to refactor the validation logic later for better readability. Ready to merge to main!
